Ruby lesson: update ruby patch version

dellaighted opened the pull request:
<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because
<!-- Summarize the purpose or reasons for this PR, e.g. what problem it solves or what benefit it provides. -->
As mentioned in this [issue](https://github.com/TheOdinProject/curriculum/issues/28403): The Ruby install lesson tells learners to install 3.3.0. but the newest patch version for Ruby with the latest security fixes is 3.3.4.

In the "Project Management" lesson, learners are told to run the command rbenv local 3.2.2 to declare the ruby version local to a project. This needs to be adjusted so learners run the command with the current version they have installed.

## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->
- changed `3.3.0` to `3.3.4` in the [Installing Ruby lesson](https://www.theodinproject.com/lessons/ruby-installing-ruby)
- slightly changed [line 329](https://github.com/TheOdinProject/curriculum/blob/main/ruby/object_oriented_programming_basics/managing_ruby_projects.md?plain=1#L329) of the [Ruby Project Management lesson](https://www.theodinproject.com/lessons/ruby-project-management)

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.

_Note:_ any pull request created for an issue that already has someone else assigned **will be closed without review**.
-->
Closes #28403

## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

dellaighted commented:
I've made the necessary changes. Didn't notice the lint errors until after pushing commit.
![image](https://github.com/user-attachments/assets/e0549ffa-fa88-4833-a7b0-51c284113f94)
I can fix the first error, but how should I fix the second error?

JoshDevHub commented on ruby_on_rails/advanced_forms_and_activerecord/forms_advanced.md:
````diff
@@ -32,7 +32,11 @@ Let's say you want to build a New Post form for your blog but you want to be abl
+```html
+=======
 ```erb
+>>>>>>> main
````
Lines 35-37 need to be deleted. Line 38 also needs to be deleted. This is all just to fully resolve the merge conflict.
```
<<<<<< HEAD
This section shows things that are in your local branch that
are different from what's on `main`. This is because your
branch had become out of date. We want to get rid of 
all of this and accept the stuff that's in `main`
======

>>>>>>> main
This stuff is content that's a part of `main`.
You want to accept these changes, but you still need
to get rid of the annotation lines `git` has placed
surrounding this content. Looks like you already got
rid of the `=======` bit, but `>>>>>> main` is still
here.
=======
```
Once you do this, this file should match `main` and will stop showing in the diff for this PR. Which is good because this file has nothing to do with the PR you're making here.

JoshDevHub reviewed (changes requested):
Don't worry about the lint issue for the heading stuff. Most installation lessons have problems with heading structure, and it's something the team plans to eventually address separately.

The other lint error should resolve itself if you fully resolve the merge conflict. Read my comment below for more about that.

MaoShizhong reviewed (commented):
Related to the heading lint errors for this PR specifically, in `installing_ruby.md`, could the first line (the disable comment) be replaced with the following 2 lines?

```md
<!-- TODO: Revisit lesson/heading structure to remove need to disable rules -->
<!-- markdownlint-disable MD024 TOP004 -->
```
Adds a TODO that other installations lesson have for us to revisit later, and also corrects the disable comment to disable the correct heading-related rule (we have a custom one, we don't use MD043)

dellaighted commented:
I fixed the error that was due to the merge conflict and left the other error alone. I ran `npm run lint:lesson -- "forms_advanced.md" and saw there were no errors so I commited and pushed all my changes. However, on github it shows there is an error in that file.

I added the comments that @MaoShizhong suggested to the `installing_ruby.md` file

MaoShizhong commented:
@dellaighted What directory was your terminal in when you ran that command?
You'll need to run that command in the root of the curriculum repo (where the `package.json` file is) so that it can pick up the config files (which include all the rules), which also means you'll need to provide the file path from that location.

So rather than navigating to `ruby_on_rails/advanced_forms_and_activerecord` then running `npm run lint:lesson -- "forms_advanced.md"`, you'll need to go back to the curriculum repo root (where `package.json` is), then run
```bash
npm run lint:lesson -- "ruby_on_rails/advanced_forms_and_activerecord/forms_advanced.md"
```

dellaighted commented:
You're right! I was running that command while in the `advanced_forms_and_activerecord` directory.
> So rather than navigating to `ruby_on_rails/advanced_forms_and_activerecord` then running `npm run lint:lesson -- "forms_advanced.md"`, you'll need to go back to the curriculum repo root (where `package.json` is), then run
> 
> ```shell
> npm run lint:lesson -- "ruby_on_rails/advanced_forms_and_activerecord/forms_advanced.md"
> ```
I went back to `curriculum` directory and ran the command there, but it shows me there's no errors.
![image](https://github.com/user-attachments/assets/42099574-7d6c-4264-b6c6-59f4c79c02af)

dellaighted commented:
Sorry I edited my previous comment to say I ran the lint script first and then the fix script. Not sure if you saw that bit.
The lint script didn't pick up any errors, so I ran the fix script to see if anything would get fixed anyways.

I just double checked the `forms_advanced.md` file, and I believe I have fixed all the errors from the merge conflict. I removed any text like `<<<<<< HEAD`. The error on github seems to be pointing to line 36 in the file.
![image](https://github.com/user-attachments/assets/09bc802e-61b7-4165-97a3-304f222598c3)

This is what it currently looks like.
![image](https://github.com/user-attachments/assets/1f2c4cf4-f64b-41ac-8e8d-d8c6229c6b55)

I ran the lint script again anyways, but it's still not picking up anything.
![image](https://github.com/user-attachments/assets/679f997c-2bde-4424-84fe-db7a3b678d99)

MaoShizhong commented:
@dellaighted Looks like you haven't committed and pushed those changes you're showing there.

If you look in the `Files changed` tab here, you'll see that `forms_advanced.md` still shows the unchanged code (with the `<<<<<< HEAD` stuff etc.). Your latest commit only includes the changes made to the first line of `installing_ruby.md` (you can see commit-specific changes by clicking on the commit itself in the list).

dellaighted commented:
Can't believe those changes didn't get pushed. Anyways I think it's all fixed now.

JoshDevHub reviewed (approved):
Looks great to me :rocket:
